Problem with test "TestTestgresCommon.test_synchronous_replication[local]"

https://github.com/dmitry-lipetsk/testgres/actions/runs/29159423824/job/86561878700

```
LOG:  could not bind IPv4 address "127.0.0.1": Address already in use
```

All log:
```
____________ TestTestgresCommon.test_synchronous_replication[local] ____________
[gw2] linux -- Python 3.7.3 /tmp/testgres_venv/bin/python3

self = PostgresNode(name='testgres-9588ccbe-4328-4a8f-a9e8-264ac76493bd', port=None, base_dir='/tmp/tgsn_1uvz2bay')
params = []

    def restart(self, params=[]):
        """
        Restart this node using pg_ctl.
    
        Args:
            params: additional arguments for pg_ctl.
    
        Returns:
            This instance of :class:`.PostgresNode`.
        """
    
        _params = [
            self._get_bin_path("pg_ctl"),
            "-D", self.data_dir,
            "-l", self.pg_log_file,
            "-w",  # wait
            "restart"
        ] + params  # yapf: disable
    
        try:
>           error_code, out, error = execute_utility2(self._os_ops, _params, self.utils_log_file, verbose=True)

_params    = ['/pg/postgres/install/bin/pg_ctl',
 '-D',
 '/tmp/tgsn_1uvz2bay/data',
 '-l',
 '/tmp/tgsn_1uvz2bay/logs/postgresql.log',
 '-w',
 'restart']
files      = [('/tmp/tgsn_1uvz2bay/data/postgresql.conf',
  b"\nfsync = off\nmax_worker_processes = 10\nlog_statement = 'all'\nlisten_"
  b"addresses = '127.0.0.1'\nport = 51440\n\nmax_replication_slots = 10\nmax"
  b'_wal_senders = 10\n\nhot_standby = on\nwal_keep_size = 320\nwal_level = '
  b"'replica'\nsynchronous_commit = remote_apply\nsynchronous_standby_names = "
  b'\'2 ("testgres-582431ad-b5c3-423e-8c9e-68d64dce35cf", "testgres-0929d20e-'
  b'cd05-40eb-9c40-7023bdac8633")\'\n'),
 ('/tmp/tgsn_1uvz2bay/data/postgresql.auto.conf',
  b'# Do not edit this file manually!\n# It will be overwritten by the ALTER '
  b'SYSTEM command.\n'),
 ('/tmp/tgsn_1uvz2bay/data/pg_hba.conf',
  b'# PostgreSQL Client Authentication Configuration File\n# ================'
  b'===================================\n#\n# Refer to the "Client Authenticat'
  b'ion" section in the PostgreSQL\n# documentation for a complete descriptio'
  b'n of this file.  A short\n# synopsis follows.\n#\n# -------------------'
  b'---\n# Authentication Records\n# ----------------------\n#\n# This file '
  b'controls: which hosts are allowed to connect, how clients\n# are authenti'
  b'cated, which PostgreSQL user names they can use, which\n# databases they '
  b'can access.  Records take one of these forms:\n#\n# local         DATABASE'
  b'  USER  METHOD  [OPTIONS]\n# host          DATABASE  USER  ADDRESS  METHO'
  b'D  [OPTIONS]\n# hostssl       DATABASE  USER  ADDRESS  METHOD  [OPTIONS]\n'
  b'# hostnossl     DATABASE  USER  ADDRESS  METHOD  [OPTIONS]\n# hostgssenc '
  b'   DATABASE  USER  ADDRESS  METHOD  [OPTIONS]\n# hostnogssenc  DATABASE  '
  b'USER  ADDRESS  METHOD  [OPTIONS]\n#\n# (The uppercase items must be replac'
  b'ed by actual values.)\n#\n# The first field is the connection type:\n# - "l'
  b'ocal" is a Unix-domain socket\n# - "host" is a TCP/IP socket (encrypted o'
  b'r not)\n# - "hostssl" is a TCP/IP socket that is SSL-encrypted\n# - "hostn'
  b'ossl" is a TCP/IP socket that is not SSL-encrypted\n# - "hostgssenc" is a'
  b' TCP/IP socket that is GSSAPI-encrypted\n# - "hostnogssenc" is a TCP/IP s'
  b'ocket that is not GSSAPI-encrypted\n#\n# DATABASE can be "all", "sameuser"'
  b', "samerole", "replication", a\n# database name, a regular expression (if'
  b' it starts with a slash (/))\n# or a comma-separated list thereof.  The "'
  b'all" keyword does not match\n# "replication".  Access to replication must'
  b' be enabled in a separate\n# record (see example below).\n#\n# USER can be '
  b'"all", a user name, a group name prefixed with "+", a\n# regular expressi'
  b'on (if it starts with a slash (/)) or a comma-separated\n# list thereof. '
  b' In both the DATABASE and USER fields you can also write\n# a file name p'
  b'refixed with "@" to include names from a separate file.\n#\n# ADDRESS spec'
  b'ifies the set of hosts the record matches.  It can be a\n# host name, or '
  b'it is made up of an IP address and a CIDR mask that is\n# an integer (bet'
  b'ween 0 and 32 (IPv4) or 128 (IPv6) inclusive) that\n# specifies the numbe'
  b'r of significant bits in the mask.  A host name\n# that starts with a dot'
  b' (.) matches a suffix of the actual host name.\n# Alternatively, you can '
  b'write an IP address and netmask in separate\n# columns to specify the set'
  b' of hosts.  Instead of a CIDR-address, you\n# can write "samehost" to mat'
  b'ch any of the server\'s own IP addresses,\n# or "samenet" to match any add'
  b'ress in any subnet that the server is\n# directly connected to.\n#\n# METHO'
  b'D can be "trust", "reject", "md5", "password", "scram-sha-256",\n# "gss",'
  b' "sspi", "ident", "peer", "pam", "ldap", "radius" or "cert".\n# Note that'
  b' "password" sends passwords in clear text; "md5" or\n# "scram-sha-256" ar'
  b'e preferred since they send encrypted passwords.\n#\n# OPTIONS are a set o'
  b'f options for the authentication in the format\n# NAME=VALUE.  The availa'
  b'ble options depend on the different\n# authentication methods -- refer to'
  b' the "Client Authentication"\n# section in the documentation for a list o'
  b'f which options are\n# available for which authentication methods.\n#\n# Da'
  b'tabase and user names containing spaces, commas, quotes and other\n# spec'
  b'ial characters must be quoted.  Quoting one of the keywords\n# "all", "sa'
  b'meuser", "samerole" or "replication" makes the name lose\n# its special c'
  b'haracter, and just match a database or username with\n# that name.\n#\n# --'
  b'-------------\n# Include Records\n# ---------------\n#\n# This file allo'
  b'ws the inclusion of external files or directories holding\n# more records'
  b', using the following keywords:\n#\n# include           FILE\n# include_if_'
  b'exists FILE\n# include_dir       DIRECTORY\n#\n# FILE is the file name to i'
  b'nclude, and DIR is the directory name containing\n# the file(s) to includ'
  b'e.  Any file in a directory will be loaded if suffixed\n# with ".conf".  '
  b'The files of a directory are ordered by name.\n# include_if_exists ignore'
  b's missing files.  FILE and DIRECTORY can be\n# specified as a relative or'
  b' an absolute path, and can be double-quoted if\n# they contain spaces.\n#\n'
  b'# -------------\n# Miscellaneous\n# -------------\n#\n# This file is rea'
  b'd on server startup and when the server receives a\n# SIGHUP signal.  If '
  b'you edit the file on a running system, you have to\n# SIGHUP the server f'
  b'or the changes to take effect, run "pg_ctl reload",\n# or execute "SELECT'
  b' pg_reload_conf()".\n#\n# ----------------------------------\n# Put your ac'
  b'tual configuration here\n# ----------------------------------\n#\n# If you '
  b'want to allow non-local connections, you need to add more\n# "host" recor'
  b'ds.  In that case you will also need to make PostgreSQL\n# listen on a no'
  b'n-local interface via the listen_addresses\n# configuration parameter, or'
  b' via the -i or -h command line switches.\n\n# CAUTION: Configuring the sys'
  b'tem for local "trust" authentication\n# allows any local user to connect '
  b'as any PostgreSQL user, including\n# the database superuser.  If you do n'
  b'ot trust all your local users,\n# use another authentication method.\n'
  b'\n\n# TYPE  DATABASE        USER            ADDRESS                 METHOD'
  b'\n\n# "local" is for Unix domain socket connections only\nlocal   all      '
  b'       all                                     trust\n# IPv4 local connec'
  b'tions:\nhost    all             all             127.0.0.1/32            t'
  b'rust\n# IPv6 local connections:\nhost    all             all             :'
  b':1/128                 trust\n# Allow replication connections from localh'
  b'ost, by a user with the\n# replication privilege.\nlocal   replication    '
  b' all                                     trust\nhost    replication     a'
  b'll             127.0.0.1/32            trust\nhost    replication     all'
  b'             ::1/128                 trust\n\n# Testgres default configura'
  b'tion\nhost     replication      all              0.0.0.0/0               '
  b' trust\nhost     replication      all              ::/0                  '
  b'   trust\nhost     all              all              0.0.0.0/0           '
  b'     trust\nhost     all              all              ::/0              '
  b'       trust\n'),
 ('/tmp/tgsn_1uvz2bay/logs/postgresql.log',
  b'2026-07-11 16:32:50.864 GMT [138041] LOG:  listening on Unix socket "/tm'
  b'p/.s.PGSQL.51440"\n2026-07-11 16:32:50.865 GMT [138045] LOG:  database sy'
  b'stem was shut down at 2026-07-11 16:32:50 GMT\n2026-07-11 16:32:50.867 GM'
  b'T [138041] LOG:  database system is ready to accept connections\n2026-07-'
  b'11 16:32:50.969 GMT [138043] LOG:  checkpoint starting: force wait\n2026-'
  b'07-11 16:32:50.973 GMT [138043] LOG:  checkpoint complete: wrote 3 buffe'
  b'rs (0.0%); 0 WAL file(s) added, 0 removed, 0 recycled; write=0.001 s, sy'
  b'nc=0.001 s, total=0.004 s; sync files=0, longest=0.000 s, average=0.000 '
  b's; distance=11417 kB, estimate=11417 kB; lsn=0/2000080, redo lsn=0/20000'
  b'28\n2026-07-11 16:32:51.464 GMT [138043] LOG:  checkpoint starting: force'
  b' wait\n2026-07-11 16:32:51.467 GMT [138043] LOG:  checkpoint complete: wr'
  b'ote 0 buffers (0.0%); 0 WAL file(s) added, 0 removed, 0 recycled; write='
  b'0.001 s, sync=0.001 s, total=0.004 s; sync files=0, longest=0.000 s, ave'
  b'rage=0.000 s; distance=16384 kB, estimate=16384 kB; lsn=0/3000080, redo '
  b'lsn=0/3000028\n2026-07-11 16:32:52.205 GMT [138041] LOG:  received fast s'
  b'hutdown request\n2026-07-11 16:32:52.205 GMT [138041] LOG:  aborting any '
  b'active transactions\n2026-07-11 16:32:52.206 GMT [138041] LOG:  backgroun'
  b'd worker "logical replication launcher" (PID 138048) exited with exit co'
  b'de 1\n2026-07-11 16:32:52.206 GMT [138043] LOG:  shutting down\n2026-07-11'
  b' 16:32:52.217 GMT [138043] LOG:  checkpoint starting: shutdown immediate'
  b'\n2026-07-11 16:32:52.220 GMT [138043] LOG:  checkpoint complete: wrote 0'
  b' buffers (0.0%); 0 WAL file(s) added, 0 removed, 0 recycled; write=0.001'
  b' s, sync=0.001 s, total=0.004 s; sync files=0, longest=0.000 s, average='
  b'0.000 s; distance=16384 kB, estimate=16384 kB; lsn=0/4000028, redo lsn=0'
  b'/4000028\n2026-07-11 16:32:52.228 GMT [138041] LOG:  database system is s'
  b'hut down\n2026-07-11 16:32:52.318 GMT [138267] LOG:  starting PostgreSQL '
  b'17.7 on x86_64-pc-linux-gnu, compiled by gcc (AstraLinuxSE 8.3.0-6) 8.3.'
  b'0, 64-bit\n2026-07-11 16:32:52.318 GMT [138267] LOG:  could not bind IPv4'
  b' address "127.0.0.1": Address already in use\n2026-07-11 16:32:52.318 GMT'
  b' [138267] HINT:  Is another postmaster already running on port 51440? If'
  b' not, wait a few seconds and retry.\n2026-07-11 16:32:52.318 GMT [138267]'
  b' WARNING:  could not create listen socket for "127.0.0.1"\n2026-07-11 16:'
  b'32:52.318 GMT [138267] FATAL:  could not create any TCP/IP sockets\n2026-'
  b'07-11 16:32:52.319 GMT [138267] LOG:  database system is shut down\n')]
msg        = 'Cannot restart node'
params     = []
self       = PostgresNode(name='testgres-9588ccbe-4328-4a8f-a9e8-264ac76493bd', port=None, base_dir='/tmp/tgsn_1uvz2bay')

src/node.py:1336: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
src/utils.py:109: in execute_utility2
    exec_env=exec_env,
        args       = ['/pg/postgres/install/bin/pg_ctl',
 '-D',
 '/tmp/tgsn_1uvz2bay/data',
 '-l',
 '/tmp/tgsn_1uvz2bay/logs/postgresql.log',
 '-w',
 'restart']
        exec_env   = None
        ignore_errors = False
        logfile    = '/tmp/tgsn_1uvz2bay/logs/utils.log'
        os_ops     = <testgres.operations.local_ops.LocalOperations object at 0x7f6ccda9f9b0>
        verbose    = True
/tmp/testgres_venv/lib/python3.7/site-packages/testgres/operations/local_ops.py:349: in exec_command
    out=run_r[1])
        cmd        = ['/pg/postgres/install/bin/pg_ctl',
 '-D',
 '/tmp/tgsn_1uvz2bay/data',
 '-l',
 '/tmp/tgsn_1uvz2bay/logs/postgresql.log',
 '-w',
 'restart']
        cwd        = None
        encoding   = 'UTF-8'
        exec_env   = None
        expect_error = False
        get_process = False
        ignore_errors = False
        input      = None
        run_r      = (1,
 'waiting for server to shut down.... done\n'
 'server stopped\n'
 'waiting for server to start.... stopped waiting\n',
 'pg_ctl: could not start server\nExamine the log output.\n')
        self       = <testgres.operations.local_ops.LocalOperations object at 0x7f6ccda9f9b0>
        shell      = False
        stderr     = None
        stdin      = None
        stdout     = None
        text       = False
        timeout    = None
        verbose    = True
        wait_exit  = False
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

cmd = ['/pg/postgres/install/bin/pg_ctl', '-D', '/tmp/tgsn_1uvz2bay/data', '-l', '/tmp/tgsn_1uvz2bay/logs/postgresql.log', '-w', ...]
exit_code = 1
msg_arg = 'pg_ctl: could not start server\nExamine the log output.\n'
error = 'pg_ctl: could not start server\nExamine the log output.\n'
out = 'waiting for server to shut down.... done\nserver stopped\nwaiting for server to start.... stopped waiting\n'

    @staticmethod
    def UtilityExitedWithNonZeroCode(cmd, exit_code, msg_arg, error, out):
        assert type(exit_code) is int
    
        msg_arg_s = __class__._TranslateDataIntoString(msg_arg)
        assert type(msg_arg_s) is str
    
        msg_arg_s = msg_arg_s.strip()
        if msg_arg_s == "":
            msg_arg_s = "#no_error_message"
    
        message = "Utility exited with non-zero code (" + str(exit_code) + "). Error: `" + msg_arg_s + "`"
        raise ExecUtilException(
            message=message,
            command=cmd,
            exit_code=exit_code,
            out=out,
>           error=error)
E       testgres.operations.exceptions.ExecUtilException: Utility exited with non-zero code (1). Error: `pg_ctl: could not start server
E       Examine the log output.`
E       Command: /pg/postgres/install/bin/pg_ctl -D /tmp/tgsn_1uvz2bay/data -l /tmp/tgsn_1uvz2bay/logs/postgresql.log -w restart
E       Exit code: 1
E       ---- Error:
E       pg_ctl: could not start server
E       Examine the log output.
E       
E       ---- Out:
E       waiting for server to shut down.... done
E       server stopped
E       waiting for server to start.... stopped waiting

__class__  = <class 'testgres.operations.raise_error.RaiseError'>
cmd        = ['/pg/postgres/install/bin/pg_ctl',
 '-D',
 '/tmp/tgsn_1uvz2bay/data',
 '-l',
 '/tmp/tgsn_1uvz2bay/logs/postgresql.log',
 '-w',
 'restart']
error      = 'pg_ctl: could not start server\nExamine the log output.\n'
exit_code  = 1
message    = ('Utility exited with non-zero code (1). Error: `pg_ctl: could not start '
 'server\n'
 'Examine the log output.`')
msg_arg    = 'pg_ctl: could not start server\nExamine the log output.\n'
msg_arg_s  = 'pg_ctl: could not start server\nExamine the log output.'
out        = ('waiting for server to shut down.... done\n'
 'server stopped\n'
 'waiting for server to start.... stopped waiting\n')

/tmp/testgres_venv/lib/python3.7/site-packages/testgres/operations/raise_error.py:23: ExecUtilException

The above exception was the direct cause of the following exception:

self = <tests.test_testgres_common.TestTestgresCommon object at 0x7f6cccb70a58>
node_svc = <tests.helpers.global_data.PostgresNodeService object at 0x7f6ccd5fcbe0>

    def test_synchronous_replication(self, node_svc: PostgresNodeService):
        assert isinstance(node_svc, PostgresNodeService)
    
        current_version = get_pg_version2(node_svc.os_ops)
    
        __class__.helper__skip_test_if_pg_version_is_not_ge(current_version, "9.6")
    
        with __class__.helper__get_node(node_svc) as master:
            old_version = not __class__.helper__pg_version_ge(current_version, '9.6')
    
            master.init(allow_streaming=True).start()
    
            if not old_version:
                master.append_conf('synchronous_commit = remote_apply')
    
            # create standby
            with master.replicate() as standby1, master.replicate() as standby2:
                standby1.start()
                standby2.start()
    
                # check formatting
                assert (
                    '1 ("{}", "{}")'.format(standby1.name, standby2.name) == str(First(1, (standby1, standby2)))
                )  # yapf: disable
                assert (
                    'ANY 1 ("{}", "{}")'.format(standby1.name, standby2.name) == str(Any(1, (standby1, standby2)))
                )  # yapf: disable
    
                # set synchronous_standby_names
                master.set_synchronous_standbys(First(2, [standby1, standby2]))
>               master.restart()

__class__  = <class 'tests.test_testgres_common.TestTestgresCommon'>
current_version = '17.7'
master     = PostgresNode(name='testgres-9588ccbe-4328-4a8f-a9e8-264ac76493bd', port=None, base_dir='/tmp/tgsn_1uvz2bay')
node_svc   = <tests.helpers.global_data.PostgresNodeService object at 0x7f6ccd5fcbe0>
old_version = False
self       = <tests.test_testgres_common.TestTestgresCommon object at 0x7f6cccb70a58>
standby1   = PostgresNode(name='testgres-582431ad-b5c3-423e-8c9e-68d64dce35cf', port=None, base_dir='/tmp/tgsb_3icqmctj')
standby2   = PostgresNode(name='testgres-0929d20e-cd05-40eb-9c40-7023bdac8633', port=None, base_dir='/tmp/tgsb_0paib0x5')

tests/test_testgres_common.py:1673: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
src/node.py:1342: in restart
    raise_from(StartNodeException(msg, files), e)
        _params    = ['/pg/postgres/install/bin/pg_ctl',
 '-D',
 '/tmp/tgsn_1uvz2bay/data',
 '-l',
 '/tmp/tgsn_1uvz2bay/logs/postgresql.log',
 '-w',
 'restart']
        files      = [('/tmp/tgsn_1uvz2bay/data/postgresql.conf',
  b"\nfsync = off\nmax_worker_processes = 10\nlog_statement = 'all'\nlisten_"
  b"addresses = '127.0.0.1'\nport = 51440\n\nmax_replication_slots = 10\nmax"
  b'_wal_senders = 10\n\nhot_standby = on\nwal_keep_size = 320\nwal_level = '
  b"'replica'\nsynchronous_commit = remote_apply\nsynchronous_standby_names = "
  b'\'2 ("testgres-582431ad-b5c3-423e-8c9e-68d64dce35cf", "testgres-0929d20e-'
  b'cd05-40eb-9c40-7023bdac8633")\'\n'),
 ('/tmp/tgsn_1uvz2bay/data/postgresql.auto.conf',
  b'# Do not edit this file manually!\n# It will be overwritten by the ALTER '
  b'SYSTEM command.\n'),
 ('/tmp/tgsn_1uvz2bay/data/pg_hba.conf',
  b'# PostgreSQL Client Authentication Configuration File\n# ================'
  b'===================================\n#\n# Refer to the "Client Authenticat'
  b'ion" section in the PostgreSQL\n# documentation for a complete descriptio'
  b'n of this file.  A short\n# synopsis follows.\n#\n# -------------------'
  b'---\n# Authentication Records\n# ----------------------\n#\n# This file '
  b'controls: which hosts are allowed to connect, how clients\n# are authenti'
  b'cated, which PostgreSQL user names they can use, which\n# databases they '
  b'can access.  Records take one of these forms:\n#\n# local         DATABASE'
  b'  USER  METHOD  [OPTIONS]\n# host          DATABASE  USER  ADDRESS  METHO'
  b'D  [OPTIONS]\n# hostssl       DATABASE  USER  ADDRESS  METHOD  [OPTIONS]\n'
  b'# hostnossl     DATABASE  USER  ADDRESS  METHOD  [OPTIONS]\n# hostgssenc '
  b'   DATABASE  USER  ADDRESS  METHOD  [OPTIONS]\n# hostnogssenc  DATABASE  '
  b'USER  ADDRESS  METHOD  [OPTIONS]\n#\n# (The uppercase items must be replac'
  b'ed by actual values.)\n#\n# The first field is the connection type:\n# - "l'
  b'ocal" is a Unix-domain socket\n# - "host" is a TCP/IP socket (encrypted o'
  b'r not)\n# - "hostssl" is a TCP/IP socket that is SSL-encrypted\n# - "hostn'
  b'ossl" is a TCP/IP socket that is not SSL-encrypted\n# - "hostgssenc" is a'
  b' TCP/IP socket that is GSSAPI-encrypted\n# - "hostnogssenc" is a TCP/IP s'
  b'ocket that is not GSSAPI-encrypted\n#\n# DATABASE can be "all", "sameuser"'
  b', "samerole", "replication", a\n# database name, a regular expression (if'
  b' it starts with a slash (/))\n# or a comma-separated list thereof.  The "'
  b'all" keyword does not match\n# "replication".  Access to replication must'
  b' be enabled in a separate\n# record (see example below).\n#\n# USER can be '
  b'"all", a user name, a group name prefixed with "+", a\n# regular expressi'
  b'on (if it starts with a slash (/)) or a comma-separated\n# list thereof. '
  b' In both the DATABASE and USER fields you can also write\n# a file name p'
  b'refixed with "@" to include names from a separate file.\n#\n# ADDRESS spec'
  b'ifies the set of hosts the record matches.  It can be a\n# host name, or '
  b'it is made up of an IP address and a CIDR mask that is\n# an integer (bet'
  b'ween 0 and 32 (IPv4) or 128 (IPv6) inclusive) that\n# specifies the numbe'
  b'r of significant bits in the mask.  A host name\n# that starts with a dot'
  b' (.) matches a suffix of the actual host name.\n# Alternatively, you can '
  b'write an IP address and netmask in separate\n# columns to specify the set'
  b' of hosts.  Instead of a CIDR-address, you\n# can write "samehost" to mat'
  b'ch any of the server\'s own IP addresses,\n# or "samenet" to match any add'
  b'ress in any subnet that the server is\n# directly connected to.\n#\n# METHO'
  b'D can be "trust", "reject", "md5", "password", "scram-sha-256",\n# "gss",'
  b' "sspi", "ident", "peer", "pam", "ldap", "radius" or "cert".\n# Note that'
  b' "password" sends passwords in clear text; "md5" or\n# "scram-sha-256" ar'
  b'e preferred since they send encrypted passwords.\n#\n# OPTIONS are a set o'
  b'f options for the authentication in the format\n# NAME=VALUE.  The availa'
  b'ble options depend on the different\n# authentication methods -- refer to'
  b' the "Client Authentication"\n# section in the documentation for a list o'
  b'f which options are\n# available for which authentication methods.\n#\n# Da'
  b'tabase and user names containing spaces, commas, quotes and other\n# spec'
  b'ial characters must be quoted.  Quoting one of the keywords\n# "all", "sa'
  b'meuser", "samerole" or "replication" makes the name lose\n# its special c'
  b'haracter, and just match a database or username with\n# that name.\n#\n# --'
  b'-------------\n# Include Records\n# ---------------\n#\n# This file allo'
  b'ws the inclusion of external files or directories holding\n# more records'
  b', using the following keywords:\n#\n# include           FILE\n# include_if_'
  b'exists FILE\n# include_dir       DIRECTORY\n#\n# FILE is the file name to i'
  b'nclude, and DIR is the directory name containing\n# the file(s) to includ'
  b'e.  Any file in a directory will be loaded if suffixed\n# with ".conf".  '
  b'The files of a directory are ordered by name.\n# include_if_exists ignore'
  b's missing files.  FILE and DIRECTORY can be\n# specified as a relative or'
  b' an absolute path, and can be double-quoted if\n# they contain spaces.\n#\n'
  b'# -------------\n# Miscellaneous\n# -------------\n#\n# This file is rea'
  b'd on server startup and when the server receives a\n# SIGHUP signal.  If '
  b'you edit the file on a running system, you have to\n# SIGHUP the server f'
  b'or the changes to take effect, run "pg_ctl reload",\n# or execute "SELECT'
  b' pg_reload_conf()".\n#\n# ----------------------------------\n# Put your ac'
  b'tual configuration here\n# ----------------------------------\n#\n# If you '
  b'want to allow non-local connections, you need to add more\n# "host" recor'
  b'ds.  In that case you will also need to make PostgreSQL\n# listen on a no'
  b'n-local interface via the listen_addresses\n# configuration parameter, or'
  b' via the -i or -h command line switches.\n\n# CAUTION: Configuring the sys'
  b'tem for local "trust" authentication\n# allows any local user to connect '
  b'as any PostgreSQL user, including\n# the database superuser.  If you do n'
  b'ot trust all your local users,\n# use another authentication method.\n'
  b'\n\n# TYPE  DATABASE        USER            ADDRESS                 METHOD'
  b'\n\n# "local" is for Unix domain socket connections only\nlocal   all      '
  b'       all                                     trust\n# IPv4 local connec'
  b'tions:\nhost    all             all             127.0.0.1/32            t'
  b'rust\n# IPv6 local connections:\nhost    all             all             :'
  b':1/128                 trust\n# Allow replication connections from localh'
  b'ost, by a user with the\n# replication privilege.\nlocal   replication    '
  b' all                                     trust\nhost    replication     a'
  b'll             127.0.0.1/32            trust\nhost    replication     all'
  b'             ::1/128                 trust\n\n# Testgres default configura'
  b'tion\nhost     replication      all              0.0.0.0/0               '
  b' trust\nhost     replication      all              ::/0                  '
  b'   trust\nhost     all              all              0.0.0.0/0           '
  b'     trust\nhost     all              all              ::/0              '
  b'       trust\n'),
 ('/tmp/tgsn_1uvz2bay/logs/postgresql.log',
  b'2026-07-11 16:32:50.864 GMT [138041] LOG:  listening on Unix socket "/tm'
  b'p/.s.PGSQL.51440"\n2026-07-11 16:32:50.865 GMT [138045] LOG:  database sy'
  b'stem was shut down at 2026-07-11 16:32:50 GMT\n2026-07-11 16:32:50.867 GM'
  b'T [138041] LOG:  database system is ready to accept connections\n2026-07-'
  b'11 16:32:50.969 GMT [138043] LOG:  checkpoint starting: force wait\n2026-'
  b'07-11 16:32:50.973 GMT [138043] LOG:  checkpoint complete: wrote 3 buffe'
  b'rs (0.0%); 0 WAL file(s) added, 0 removed, 0 recycled; write=0.001 s, sy'
  b'nc=0.001 s, total=0.004 s; sync files=0, longest=0.000 s, average=0.000 '
  b's; distance=11417 kB, estimate=11417 kB; lsn=0/2000080, redo lsn=0/20000'
  b'28\n2026-07-11 16:32:51.464 GMT [138043] LOG:  checkpoint starting: force'
  b' wait\n2026-07-11 16:32:51.467 GMT [138043] LOG:  checkpoint complete: wr'
  b'ote 0 buffers (0.0%); 0 WAL file(s) added, 0 removed, 0 recycled; write='
  b'0.001 s, sync=0.001 s, total=0.004 s; sync files=0, longest=0.000 s, ave'
  b'rage=0.000 s; distance=16384 kB, estimate=16384 kB; lsn=0/3000080, redo '
  b'lsn=0/3000028\n2026-07-11 16:32:52.205 GMT [138041] LOG:  received fast s'
  b'hutdown request\n2026-07-11 16:32:52.205 GMT [138041] LOG:  aborting any '
  b'active transactions\n2026-07-11 16:32:52.206 GMT [138041] LOG:  backgroun'
  b'd worker "logical replication launcher" (PID 138048) exited with exit co'
  b'de 1\n2026-07-11 16:32:52.206 GMT [138043] LOG:  shutting down\n2026-07-11'
  b' 16:32:52.217 GMT [138043] LOG:  checkpoint starting: shutdown immediate'
  b'\n2026-07-11 16:32:52.220 GMT [138043] LOG:  checkpoint complete: wrote 0'
  b' buffers (0.0%); 0 WAL file(s) added, 0 removed, 0 recycled; write=0.001'
  b' s, sync=0.001 s, total=0.004 s; sync files=0, longest=0.000 s, average='
  b'0.000 s; distance=16384 kB, estimate=16384 kB; lsn=0/4000028, redo lsn=0'
  b'/4000028\n2026-07-11 16:32:52.228 GMT [138041] LOG:  database system is s'
  b'hut down\n2026-07-11 16:32:52.318 GMT [138267] LOG:  starting PostgreSQL '
  b'17.7 on x86_64-pc-linux-gnu, compiled by gcc (AstraLinuxSE 8.3.0-6) 8.3.'
  b'0, 64-bit\n2026-07-11 16:32:52.318 GMT [138267] LOG:  could not bind IPv4'
  b' address "127.0.0.1": Address already in use\n2026-07-11 16:32:52.318 GMT'
  b' [138267] HINT:  Is another postmaster already running on port 51440? If'
  b' not, wait a few seconds and retry.\n2026-07-11 16:32:52.318 GMT [138267]'
  b' WARNING:  could not create listen socket for "127.0.0.1"\n2026-07-11 16:'
  b'32:52.318 GMT [138267] FATAL:  could not create any TCP/IP sockets\n2026-'
  b'07-11 16:32:52.319 GMT [138267] LOG:  database system is shut down\n')]
        msg        = 'Cannot restart node'
        params     = []
        self       = PostgresNode(name='testgres-9588ccbe-4328-4a8f-a9e8-264ac76493bd', port=None, base_dir='/tmp/tgsn_1uvz2bay')
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

value = None
from_value = ExecUtilException(message='Utility exited with non-zero code (1). Error: `pg_ctl: could not start server\nExamine the ...nwaiting for server to start.... stopped waiting\n', error='pg_ctl: could not start server\nExamine the log output.\n')

>   ???
E   src.exceptions.StartNodeException: Cannot restart node
E   /tmp/tgsn_1uvz2bay/data/postgresql.conf
E   ----
E   b'\nfsync = off\nmax_worker_processes = 10\nlog_statement = \'all\'\nlisten_addresses = \'127.0.0.1\'\nport = 51440\n\nmax_replication_slots = 10\nmax_wal_senders = 10\n\nhot_standby = on\nwal_keep_size = 320\nwal_level = \'replica\'\nsynchronous_commit = remote_apply\nsynchronous_standby_names = \'2 ("testgres-582431ad-b5c3-423e-8c9e-68d64dce35cf", "testgres-0929d20e-cd05-40eb-9c40-7023bdac8633")\'\n'
E   
E   /tmp/tgsn_1uvz2bay/data/postgresql.auto.conf
E   ----
E   b'# Do not edit this file manually!\n# It will be overwritten by the ALTER SYSTEM command.\n'
E   
E   /tmp/tgsn_1uvz2bay/data/pg_hba.conf
E   ----
E   b'# PostgreSQL Client Authentication Configuration File\n# ===================================================\n#\n# Refer to the "Client Authentication" section in the PostgreSQL\n# documentation for a complete description of this file.  A short\n# synopsis follows.\n#\n# ----------------------\n# Authentication Records\n# ----------------------\n#\n# This file controls: which hosts are allowed to connect, how clients\n# are authenticated, which PostgreSQL user names they can use, which\n# databases they can access.  Records take one of these forms:\n#\n# local         DATABASE  USER  METHOD  [OPTIONS]\n# host          DATABASE  USER  ADDRESS  METHOD  [OPTIONS]\n# hostssl       DATABASE  USER  ADDRESS  METHOD  [OPTIONS]\n# hostnossl     DATABASE  USER  ADDRESS  METHOD  [OPTIONS]\n# hostgssenc    DATABASE  USER  ADDRESS  METHOD  [OPTIONS]\n# hostnogssenc  DATABASE  USER  ADDRESS  METHOD  [OPTIONS]\n#\n# (The uppercase items must be replaced by actual values.)\n#\n# The first field is the connection type:\n# - "local" is a Unix-domain socket\n# - "host" is a TCP/IP socket (encrypted or not)\n# - "hostssl" is a TCP/IP socket that is SSL-encrypted\n# - "hostnossl" is a TCP/IP socket that is not SSL-encrypted\n# - "hostgssenc" is a TCP/IP socket that is GSSAPI-encrypted\n# - "hostnogssenc" is a TCP/IP socket that is not GSSAPI-encrypted\n#\n# DATABASE can be "all", "sameuser", "samerole", "replication", a\n# database name, a regular expression (if it starts with a slash (/))\n# or a comma-separated list thereof.  The "all" keyword does not match\n# "replication".  Access to replication must be enabled in a separate\n# record (see example below).\n#\n# USER can be "all", a user name, a group name prefixed with "+", a\n# regular expression (if it starts with a slash (/)) or a comma-separated\n# list thereof.  In both the DATABASE and USER fields you can also write\n# a file name prefixed with "@" to include names from a separate file.\n#\n# ADDRESS specifies the set of hosts the record matches.  It can be a\n# host name, or it is made up of an IP address and a CIDR mask that is\n# an integer (between 0 and 32 (IPv4) or 128 (IPv6) inclusive) that\n# specifies the number of significant bits in the mask.  A host name\n# that starts with a dot (.) matches a suffix of the actual host name.\n# Alternatively, you can write an IP address and netmask in separate\n# columns to specify the set of hosts.  Instead of a CIDR-address, you\n# can write "samehost" to match any of the server\'s own IP addresses,\n# or "samenet" to match any address in any subnet that the server is\n# directly connected to.\n#\n# METHOD can be "trust", "reject", "md5", "password", "scram-sha-256",\n# "gss", "sspi", "ident", "peer", "pam", "ldap", "radius" or "cert".\n# Note that "password" sends passwords in clear text; "md5" or\n# "scram-sha-256" are preferred since they send encrypted passwords.\n#\n# OPTIONS are a set of options for the authentication in the format\n# NAME=VALUE.  The available options depend on the different\n# authentication methods -- refer to the "Client Authentication"\n# section in the documentation for a list of which options are\n# available for which authentication methods.\n#\n# Database and user names containing spaces, commas, quotes and other\n# special characters must be quoted.  Quoting one of the keywords\n# "all", "sameuser", "samerole" or "replication" makes the name lose\n# its special character, and just match a database or username with\n# that name.\n#\n# ---------------\n# Include Records\n# ---------------\n#\n# This file allows the inclusion of external files or directories holding\n# more records, using the following keywords:\n#\n# include           FILE\n# include_if_exists FILE\n# include_dir       DIRECTORY\n#\n# FILE is the file name to include, and DIR is the directory name containing\n# the file(s) to include.  Any file in a directory will be loaded if suffixed\n# with ".conf".  The files of a directory are ordered by name.\n# include_if_exists ignores missing files.  FILE and DIRECTORY can be\n# specified as a relative or an absolute path, and can be double-quoted if\n# they contain spaces.\n#\n# -------------\n# Miscellaneous\n# -------------\n#\n# This file is read on server startup and when the server receives a\n# SIGHUP signal.  If you edit the file on a running system, you have to\n# SIGHUP the server for the changes to take effect, run "pg_ctl reload",\n# or execute "SELECT pg_reload_conf()".\n#\n# ----------------------------------\n# Put your actual configuration here\n# ----------------------------------\n#\n# If you want to allow non-local connections, you need to add more\n# "host" records.  In that case you will also need to make PostgreSQL\n# listen on a non-local interface via the listen_addresses\n# configuration parameter, or via the -i or -h command line switches.\n\n# CAUTION: Configuring the system for local "trust" authentication\n# allows any local user to connect as any PostgreSQL user, including\n# the database superuser.  If you do not trust all your local users,\n# use another authentication method.\n\n\n# TYPE  DATABASE        USER            ADDRESS                 METHOD\n\n# "local" is for Unix domain socket connections only\nlocal   all             all                                     trust\n# IPv4 local connections:\nhost    all             all             127.0.0.1/32            trust\n# IPv6 local connections:\nhost    all             all             ::1/128                 trust\n# Allow replication connections from localhost, by a user with the\n# replication privilege.\nlocal   replication     all                                     trust\nhost    replication     all             127.0.0.1/32            trust\nhost    replication     all             ::1/128                 trust\n\n# Testgres default configuration\nhost     replication      all              0.0.0.0/0                trust\nhost     replication      all              ::/0                     trust\nhost     all              all              0.0.0.0/0                trust\nhost     all              all              ::/0                     trust\n'
E   
E   /tmp/tgsn_1uvz2bay/logs/postgresql.log
E   ----
E   b'2026-07-11 16:32:50.864 GMT [138041] LOG:  listening on Unix socket "/tmp/.s.PGSQL.51440"\n2026-07-11 16:32:50.865 GMT [138045] LOG:  database system was shut down at 2026-07-11 16:32:50 GMT\n2026-07-11 16:32:50.867 GMT [138041] LOG:  database system is ready to accept connections\n2026-07-11 16:32:50.969 GMT [138043] LOG:  checkpoint starting: force wait\n2026-07-11 16:32:50.973 GMT [138043] LOG:  checkpoint complete: wrote 3 buffers (0.0%); 0 WAL file(s) added, 0 removed, 0 recycled; write=0.001 s, sync=0.001 s, total=0.004 s; sync files=0, longest=0.000 s, average=0.000 s; distance=11417 kB, estimate=11417 kB; lsn=0/2000080, redo lsn=0/2000028\n2026-07-11 16:32:51.464 GMT [138043] LOG:  checkpoint starting: force wait\n2026-07-11 16:32:51.467 GMT [138043] LOG:  checkpoint complete: wrote 0 buffers (0.0%); 0 WAL file(s) added, 0 removed, 0 recycled; write=0.001 s, sync=0.001 s, total=0.004 s; sync files=0, longest=0.000 s, average=0.000 s; distance=16384 kB, estimate=16384 kB; lsn=0/3000080, redo lsn=0/3000028\n2026-07-11 16:32:52.205 GMT [138041] LOG:  received fast shutdown request\n2026-07-11 16:32:52.205 GMT [138041] LOG:  aborting any active transactions\n2026-07-11 16:32:52.206 GMT [138041] LOG:  background worker "logical replication launcher" (PID 138048) exited with exit code 1\n2026-07-11 16:32:52.206 GMT [138043] LOG:  shutting down\n2026-07-11 16:32:52.217 GMT [138043] LOG:  checkpoint starting: shutdown immediate\n2026-07-11 16:32:52.220 GMT [138043] LOG:  checkpoint complete: wrote 0 buffers (0.0%); 0 WAL file(s) added, 0 removed, 0 recycled; write=0.001 s, sync=0.001 s, total=0.004 s; sync files=0, longest=0.000 s, average=0.000 s; distance=16384 kB, estimate=16384 kB; lsn=0/4000028, redo lsn=0/4000028\n2026-07-11 16:32:52.228 GMT [138041] LOG:  database system is shut down\n2026-07-11 16:32:52.318 GMT [138267] LOG:  starting PostgreSQL 17.7 on x86_64-pc-linux-gnu, compiled by gcc (AstraLinuxSE 8.3.0-6) 8.3.0, 64-bit\n2026-07-11 16:32:52.318 GMT [138267] LOG:  could not bind IPv4 address "127.0.0.1": Address already in use\n2026-07-11 16:32:52.318 GMT [138267] HINT:  Is another postmaster already running on port 51440? If not, wait a few seconds and retry.\n2026-07-11 16:32:52.318 GMT [138267] WARNING:  could not create listen socket for "127.0.0.1"\n2026-07-11 16:32:52.318 GMT [138267] FATAL:  could not create any TCP/IP sockets\n2026-07-11 16:32:52.319 GMT [138267] LOG:  database system is shut down\n'

from_value = ExecUtilException(message='Utility exited with non-zero code (1). Error: `pg_ctl: could not start server\nExamine the log output.`', command=['/pg/postgres/install/bin/pg_ctl', '-D', '/tmp/tgsn_1uvz2bay/data', '-l', '/tmp/tgsn_1uvz2bay/logs/postgresql.log', '-w', 'restart'], exit_code=1, out='waiting for server to shut down.... done\nserver stopped\nwaiting for server to start.... stopped waiting\n', error='pg_ctl: could not start server\nExamine the log output.\n')
value      = None

<string>:3: StartNodeException
------------------------------ Captured log call -------------------------------
DEBUG    root:port_manager__generic.py:97 [port manager] Port 51440 is reserved.
DEBUG    root:port_manager__generic.py:97 [port manager] Port 53582 is reserved.
DEBUG    root:port_manager__generic.py:97 [port manager] Port 8450 is reserved.
DEBUG    root:port_manager__generic.py:97 [port manager] Port 8450 is released.
DEBUG    root:port_manager__generic.py:97 [port manager] Port 53582 is released.
DEBUG    root:port_manager__generic.py:97 [port manager] Port 51440 is released.
=========================== short test summary info ============================
FAILED tests/test_testgres_common.py::TestTestgresCommon::test_synchronous_replication[local] - src.exceptions.StartNodeException: Cannot restart node
======= 1 failed, 557 passed, 3 skipped, 1 xfailed in 581.24s (0:09:41) ========
```

Artefacts:
[testgres--test_logs--20260711_161528-astralinux_1_7-py3-pg17-id29159423824.zip](https://github.com/user-attachments/files/29927853/testgres--test_logs--20260711_161528-astralinux_1_7-py3-pg17-id29159423824.zip)

https://github.com/postgrespro/testgres/actions/runs/29559424756/job/87818593802

```
FAILED tests/test_testgres_common.py::TestTestgresCommon::test_synchronous_replication[remote] - src.exceptions.StartNodeException: Cannot restart node
/tmp/tgsn_Tpwlsk/data/postgresql.conf
----
b'\nfsync = off\nmax_worker_processes = 10\nlog_statement = \'all\'\nlisten_addresses = \'machine2-alpine-py3_12_xx-pg14_xx\'\nport = 12172\n\nmax_replication_slots = 10\nmax_wal_senders = 10\n\nhot_standby = on\nwal_keep_size = 320\nwal_level = \'replica\'\nsynchronous_commit = remote_apply\nsynchronous_standby_names = \'2 ("testgres-3f8f51d5-ca99-4e9f-8452-8415517fa426", "testgres-c9332c95-48ce-4607-98ae-a4052ac811aa")\'\n'

/tmp/tgsn_Tpwlsk/data/postgresql.auto.conf
----
b'# Do not edit this file manually!\n# It will be overwritten by the ALTER SYSTEM command.\n'

/tmp/tgsn_Tpwlsk/data/pg_hba.conf
----
b'# PostgreSQL Client Authentication Configuration File\n# ===================================================\n#\n# Refer to the "Client Authentication" section in the PostgreSQL\n# documentation for a complete description of this file.  A short\n# synopsis follows.\n#\n# This file controls: which hosts are allowed to connect, how clients\n# are authenticated, which PostgreSQL user names they can use, which\n# databases they can access.  Records take one of these forms:\n#\n# local         DATABASE  USER  METHOD  [OPTIONS]\n# host          DATABASE  USER  ADDRESS  METHOD  [OPTIONS]\n# hostssl       DATABASE  USER  ADDRESS  METHOD  [OPTIONS]\n# hostnossl     DATABASE  USER  ADDRESS  METHOD  [OPTIONS]\n# hostgssenc    DATABASE  USER  ADDRESS  METHOD  [OPTIONS]\n# hostnogssenc  DATABASE  USER  ADDRESS  METHOD  [OPTIONS]\n#\n# (The uppercase items must be replaced by actual values.)\n#\n# The first field is the connection type:\n# - "local" is a Unix-domain socket\n# - "host" is a TCP/IP socket (encrypted or not)\n# - "hostssl" is a TCP/IP socket that is SSL-encrypted\n# - "hostnossl" is a TCP/IP socket that is not SSL-encrypted\n# - "hostgssenc" is a TCP/IP socket that is GSSAPI-encrypted\n# - "hostnogssenc" is a TCP/IP socket that is not GSSAPI-encrypted\n#\n# DATABASE can be "all", "sameuser", "samerole", "replication", a\n# database name, or a comma-separated list thereof. The "all"\n# keyword does not match "replication". Access to replication\n# must be enabled in a separate record (see example below).\n#\n# USER can be "all", a user name, a group name prefixed with "+", or a\n# comma-separated list thereof.  In both the DATABASE and USER fields\n# you can also write a file name prefixed with "@" to include names\n# from a separate file.\n#\n# ADDRESS specifies the set of hosts the record matches.  It can be a\n# host name, or it is made up of an IP address and a CIDR mask that is\n# an integer (between 0 and 32 (IPv4) or 128 (IPv6) inclusive) that\n# specifies the number of significant bits in the mask.  A host name\n# that starts with a dot (.) matches a suffix of the actual host name.\n# Alternatively, you can write an IP address and netmask in separate\n# columns to specify the set of hosts.  Instead of a CIDR-address, you\n# can write "samehost" to match any of the server\'s own IP addresses,\n# or "samenet" to match any address in any subnet that the server is\n# directly connected to.\n#\n# METHOD can be "trust", "reject", "md5", "password", "scram-sha-256",\n# "gss", "sspi", "ident", "peer", "pam", "ldap", "radius" or "cert".\n# Note that "password" sends passwords in clear text; "md5" or\n# "scram-sha-256" are preferred since they send encrypted passwords.\n#\n# OPTIONS are a set of options for the authentication in the format\n# NAME=VALUE.  The available options depend on the different\n# authentication methods -- refer to the "Client Authentication"\n# section in the documentation for a list of which options are\n# available for which authentication methods.\n#\n# Database and user names containing spaces, commas, quotes and other\n# special characters must be quoted.  Quoting one of the keywords\n# "all", "sameuser", "samerole" or "replication" makes the name lose\n# its special character, and just match a database or username with\n# that name.\n#\n# This file is read on server startup and when the server receives a\n# SIGHUP signal.  If you edit the file on a running system, you have to\n# SIGHUP the server for the changes to take effect, run "pg_ctl reload",\n# or execute "SELECT pg_reload_conf()".\n#\n# Put your actual configuration here\n# ----------------------------------\n#\n# If you want to allow non-local connections, you need to add more\n# "host" records.  In that case you will also need to make PostgreSQL\n# listen on a non-local interface via the listen_addresses\n# configuration parameter, or via the -i or -h command line switches.\n\n# CAUTION: Configuring the system for local "trust" authentication\n# allows any local user to connect as any PostgreSQL user, including\n# the database superuser.  If you do not trust all your local users,\n# use another authentication method.\n\n\n# TYPE  DATABASE        USER            ADDRESS                 METHOD\n\n# "local" is for Unix domain socket connections only\nlocal   all             all                                     trust\n# IPv4 local connections:\nhost    all             all             127.0.0.1/32            trust\n# IPv6 local connections:\nhost    all             all             ::1/128                 trust\n# Allow replication connections from localhost, by a user with the\n# replication privilege.\nlocal   replication     all                                     trust\nhost    replication     all             127.0.0.1/32            trust\nhost    replication     all             ::1/128                 trust\n\n# Testgres default configuration\nhost     replication      all              0.0.0.0/0                trust\nhost     replication      all              ::/0                     trust\nhost     all              all              0.0.0.0/0                trust\nhost     all              all              ::/0                     trust\n'

/tmp/tgsn_Tpwlsk/logs/postgresql.log
----
b'2026-07-17 06:40:32.642 GMT [70299] LOG:  starting PostgreSQL 14.23 on x86_64-pc-linux-musl, compiled by gcc (Alpine 15.2.0) 15.2.0, 64-bit\n2026-07-17 06:40:32.642 GMT [70299] LOG:  listening on IPv4 address "172.18.0.2", port 12172\n2026-07-17 06:40:32.643 GMT [70299] LOG:  listening on Unix socket "/var/run/postgresql/.s.PGSQL.12172"\n2026-07-17 06:40:32.644 GMT [70300] LOG:  database system was shut down at 2026-07-17 06:40:32 GMT\n2026-07-17 06:40:32.648 GMT [70299] LOG:  database system is ready to accept connections\n2026-07-17 06:40:34.766 GMT [70299] LOG:  received fast shutdown request\n2026-07-17 06:40:34.766 GMT [70299] LOG:  aborting any active transactions\n2026-07-17 06:40:34.768 GMT [70299] LOG:  background worker "logical replication launcher" (PID 70306) exited with exit code 1\n2026-07-17 06:40:34.774 GMT [70301] LOG:  shutting down\n2026-07-17 06:40:34.808 GMT [70299] LOG:  database system is shut down\n2026-07-17 06:40:34.881 GMT [70565] LOG:  starting PostgreSQL 14.23 on x86_64-pc-linux-musl, compiled by gcc (Alpine 15.2.0) 15.2.0, 64-bit\n2026-07-17 06:40:34.882 GMT [70565] LOG:  could not bind IPv4 address "172.18.0.2": Address in use\n2026-07-17 06:40:34.882 GMT [70565] HINT:  Is another postmaster already running on port 12172? If not, wait a few seconds and retry.\n2026-07-17 06:40:34.882 GMT [70565] WARNING:  could not create listen socket for "machine2-alpine-py3_12_xx-pg14_xx"\n2026-07-17 06:40:34.882 GMT [70565] FATAL:  could not create any TCP/IP sockets\n2026-07-17 06:40:34.883 GMT [70565] LOG:  database system is shut down\n'
= 1 failed, 1291 passed, 3 skipped, 1 xfailed, 1 warning in 417.31s (0:06:57) ==
```

--- a/tests/test_testgres_common.py
+++ b/tests/test_testgres_common.py
@@ -1726,12 +1726,9 @@ class TestTestgresCommon:
         __class__.helper__skip_test_if_pg_version_is_not_ge(current_version, "9.6")
 
         with __class__.helper__get_node(node_svc) as master:
-            old_version = not __class__.helper__pg_version_ge(current_version, '9.6')
-
             master.init(allow_streaming=True).start()
 
-            if not old_version:
-                master.append_conf('synchronous_commit = remote_apply')
+            master.append_conf('synchronous_commit = remote_apply')
 
             # create standby
             with master.replicate() as standby1, master.replicate() as standby2:
@@ -1748,21 +1745,20 @@ class TestTestgresCommon:
 
                 # set synchronous_standby_names
                 master.set_synchronous_standbys(First(2, [standby1, standby2]))
-                master.restart()
+                master.reload()
 
-                # the following part of the test is only applicable to newer
-                # versions of PostgresQL
-                if not old_version:
-                    master.safe_psql('create table abc(a int)')
+                master.safe_psql('create table abc(a int)')
 
-                    # Create a large transaction that will take some time to apply
-                    # on standby to check that it applies synchronously
-                    # (If set synchronous_commit to 'on' or other lower level then
-                    # standby most likely won't catchup so fast and test will fail)
-                    master.safe_psql(
-                        'insert into abc select generate_series(1, 1000000)')
-                    res = standby1.safe_psql('select count(*) from abc')
-                    assert (__class__.helper__rm_carriage_returns(res) == b'1000000\n')
+                # Create a large transaction that will take some time to apply
+                # on standby to check that it applies synchronously
+                # (If set synchronous_commit to 'on' or other lower level then
+                # standby most likely won't catchup so fast and test will fail)
+                master.safe_psql(
+                    'insert into abc select generate_series(1, 1000000)',
+                )
+                res = standby1.safe_psql('select count(*) from abc')
+                assert (__class__.helper__rm_carriage_returns(res) == b'1000000\n')
+        return
 
     def test_logical_replication(self, node_svc: PostgresNodeService):
         assert isinstance(node_svc, PostgresNodeService)

